### DISCUSSION TITLE
Got different hash when run `webpack --watch --mode production` on different clones of the project?

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#15754] [Q&A]
As title. I cloned the same project into another directory on my computer, and I tried to build it in this new location. It turns out the hash is different than the old one. Why? Some of the files of the project are git-ignored, so I have to copy-paste these files into this new location. I guess the hash is different because these files have a different created dates and this info is used when generating the hash?

**Comment by alexander-akait** [comment]
What is `context`?

**Reply by nyngwang** [reply]
```js
const pRoot = path.dirname(fileURLToPath(import.meta.url))
// ...
  return {
    context: path.resolve(pRoot),
    // ...
```

**Reply by nyngwang** [reply]
it makes sense if `context` is used in calculating the hash.

**Reply by alexander-akait** [reply]
Do you use macos? There are problems with case insensivity, if you have `import foo from "./Foo.js";`, it will be different on macos and windows/linux

**Reply by nyngwang** [reply]
I have both clones on the same computer, macOS 12.3.1. So am I correct on this?

> it makes sense if `context` is used in calculating the hash.

**Reply by alexander-akait** [reply]
hm, weird, maybe you can provide for which files they are different? Maybe screenshot?

**Reply by nyngwang** [reply]
<img src="https://user-images.githubusercontent.com/24765272/166393105-afb2d368-d72f-4941-96e0-c071933545f7.png" width="300">

where `v/` is the folder for vendor files. It's strange that only `index.html` and the `main.[hash].bundle.js` changed

**Reply by alexander-akait** [reply]
Maybe you have loader/plugin which insert - time, hash, path to file, etc, can you look at source code between different versions?

**Reply by vankop** [reply]
@nyngwang please create a reproducible repo ( try to reduce repo as much as possible ). I will take a look..

**Reply by nyngwang** [reply]
@vankop Then it will take time. I encountered this problem when working. The project is not sharable. I think this problem is easily reproducible: just create/clone a front-end project using Webpack, where one at `~/SomeFolder`, and one at `~/Desktop/Another/Folder`.
